src/odemis/acq/align/fastem.py

```diff
@@ -18,22 +20,288 @@
+    :param det_rotator: (actuator) K-mirror controller. Must have a rotational (rz) axis.
+    :param calibrations: (list) List of calibrations that should be run.
+
+    :returns: (ProgressiveFuture) Alignment future object, which can be cancelled.
+    """
+
+    if not fastem_calibrations:
+        raise ModuleNotFoundError("fastem_calibration module missing. Cannot run calibrations.")
+
+    est_dur = estimate_calibration_time(calibrations)
+    f = model.ProgressiveFuture(start=time.time(), end=time.time() + est_dur)
+
+    # Create a task that runs the calibration and alignments.
+    task = CalibrationTask(f, scanner, multibeam, descanner, detector, stage, ccd, beamshift, det_rotator, calibrations)
+
+    f.task_canceller = task.cancel  # lets the future know how to cancel the task.
+
+    # Connect the future to the task and run it in a thread.
+    # task.run is executed by the executor and runs as soon as no other task is executed
+    _executor.submitf(f, task.run)
+
     return f
 
 
-def _run_fake_alignment():
-    time.sleep(2)
+def estimate_calibration_time(calibrations):
+    """
+    Computes the approximate time it will take to run all calibrations.
+    :param calibrations: (list) List of calibrations that should be run.
+    :return (0 <= float): The estimated time for the requested calibrations in s.
+    """
+
+    return sum(c.value.estimate_calibration_time() for c in calibrations)
+
+
+class CalibrationTask(object):
+    """
+    The calibration task, which runs the calibrations according to the order in the list of calibrations passed.
+    """
+
+    def __init__(self, future, scanner, multibeam, descanner, detector, stage, ccd, beamshift, det_rotator,
+                 calibrations):
+        """
+        :param future: (ProgressiveFuture) Acquisition future object, which can be cancelled.
+                       (Exception or None): Exception raised during the calibration or None.
+        :param scanner: (xt_client.Scanner) Scanner component connecting to the XT adapter.
+        :param multibeam: (technolution.EBeamScanner) The multibeam scanner component of the acquisition server module.
+        :param descanner: (technolution.MirrorDescanner) The mirror descanner component of the acquisition server
+            module.
+        :param detector: (technolution.MPPC) The detector object to be used for collecting the image data.
+        :param stage: (actuator) The stage in the corrected scan coordinate system, the x and y axes are
+            aligned with the x and y axes of the multiprobe and the multibeam scanner. Must have x, y and z axes.
+        :param ccd: (model.DigitalCamera) A camera object of the diagnostic camera.
+        :param beamshift: (tfsbc.BeamShiftController) Component that controls the beamshift deflection.
+        :param det_rotator: (actuator) K-mirror controller. Must have a rotational (rz) axis.
+        :param calibrations: (list) List of calibrations that should be run.
```

Comment:
Same here: Shouldn't this now be a Calibrations Enum object instead of a list?

Or is it a list of enumeration members from the Calibrations object? (I think that's the correct nomenclature, see: https://docs.python.org/3/library/enum.html#creating-an-enum )

Reply:
The second one should be correct, as it is still a list that is passed.

Reply:
How about this:
`:param calibrations: (list enumeration members from the Calibrations object) List of calibrations that should be run.`

Reply:
Thanks! Can you add this to the docstring?

Reply:
sounds good

Reply:
I think the type can be just `list[Calibrations]` or `list of Calibrations`.